components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -130,8 +130,14 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
       chrome.braveRewards.getWalletProperties()
       break
     case types.ON_WALLET_PROPERTIES: {
-      state = { ...state }
-      state.walletProperties = payload.properties
+      state = {
+        ...state,
+        walletCreated: true,
+        walletCreateFailed: false,
+        walletCreating: false,
+        walletCorrupted: false,
```

Comment:
@NejcZdovc @ryanml Is it safe to assume these boolean wallet properties every time we receive this action? Can we test that corruption and failure still are correctly reduced?

Reply:
I needed to add this as without it we never get `walletCreated: true` since it happens before the extension is started.

Reply:
@petemill the properties payload is typed out here: https://github.com/brave/brave-core/blob/master/components/definitions/rewardsExtensions.d.ts#L109

It is just an optional array of grants, so it should be safe to assume these

Reply:
we can't assume this. Wallet properties result could still be returned even if wallet is corrupted for a contribution . Everything else is safe to set except `walletCorrupted`. Can you please give me some more background why do we need to do this?

Reply:
@NejcZdovc because previous to this PR, the only time that walletCreated was set to true (and walletCorrupted change back to false if it was true before) was in `ON_WALLET_INITIALIZED`. However, if the extension starts _after_ the wallet is first initialized, it never received this event and AFAIK we do not have a `getWalletStatus` or something else that would return `RewardsExtension.Result.WALLET_CORRUPT`, etc like we do in https://github.com/brave/brave-core/blob/b178a589ac96e51bd89a73be0a80d59828f5aad9/components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts#L21-L53

Reply:
all good for now, we will add new API at the later stage